### PR TITLE
Fix step 1 of Range::InsertNode

### PR DESCRIPTION
**Pull request by nox**
The method needs to throw when trying to insert start node into range.

[Review on Reviewable](https://reviewable.io/reviews/servo/servo/9797)

### CODE REVIEW

File: components/script/dom/range.rs [hunk 1]
```diff
@@ -658,6 +658,9 @@ impl RangeMethods for Range {
         let start_offset = self.StartOffset();
 
         // Step 1.
+        if &*start_node == node {
+            return Err(Error::HierarchyRequest);
+        }
         match start_node.type_id() {
             // Handled under step 2.
             NodeTypeId::CharacterData(CharacterDataTypeId::Text) => (),
```
Comment (KiChjang) [inline components/script/dom/range.rs]: This probably needs fixing as well, according to the spec.
> If range’s start node is a ProcessingInstruction or Comment node, is a Text node whose parent is null, or is node, throw an HierarchyRequestError exception.

It needs to check whether start's parent is null.

Comment (nox) [inline components/script/dom/range.rs]: This is already covered in step 2, as mentioned in the comment above the line you commented. :)

**Comment by nox** [comment]
That also fixes all the remaining failures in `/dom/ranges/Range-surroundContents.html`.

@bors-servo r=KiChjang

**Pull request merged**